### ISSUE TITLE
Flux unit conversion not working properly

### ISSUE THREAD

**Issue by hcferguson** [spacetelescope/specviz#162]
Tried to change the flux units on example_spec_ipac.txt to Jy. Got the traceback below. The following incantation works in Astropy (are you using u.spectral_density for this axis?):
```
s = Table.read('example_spec_ipac.txt',format='ascii.ipac')
f = s['flux']
ff = f.to('Jy',equivalencies=u.spectral_density(s['wavelength']))
```

Here's the traceback:
```
(specviz_env)Henrys-MacBook-Pro:specview hcferguson$ specviz
Traceback (most recent call last):
  File "/Users/hcferguson/anaconda/envs/specviz_env/lib/python3.5/site-packages/astropy/units/core.py", line 866, in _get_converter
    scale = self._to(other)
  File "/Users/hcferguson/anaconda/envs/specviz_env/lib/python3.5/site-packages/astropy/units/core.py", line 935, in _to
    "'{0!r}' is not a scaled version of '{1!r}'".format(self, other))
astropy.units.core.UnitConversionError: 'Unit("erg / (Angstrom cm2 s)")' is not a scaled version of 'Unit("Jy")'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/specviz/build/lib.macosx-10.5-x86_64-3.5/specviz/ui/widgets/sub_windows.py", line 105, in _show_unit_change_dialog
    self.change_units(x_unit, y_unit)
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/specviz/build/lib.macosx-10.5-x86_64-3.5/specviz/ui/widgets/sub_windows.py", line 223, in change_units
    cntr.change_units(x, y, z)
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/specviz/build/lib.macosx-10.5-x86_64-3.5/specviz/core/containers.py", line 59, in change_units
    self.update()
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/specviz/build/lib.macosx-10.5-x86_64-3.5/specviz/core/containers.py", line 139, in update
    self.data.value)
  File "/Users/hcferguson/Documents/stsci/science_software/data_analysis_tools/sprints/specview/specviz/build/lib.macosx-10.5-x86_64-3.5/specviz/core/containers.py", line 87, in data
    return self.layer.data.to(self._plot_units[1])
  File "/Users/hcferguson/anaconda/envs/specviz_env/lib/python3.5/site-packages/astropy/units/quantity.py", line 634, in to
    equivalencies=equivalencies)
  File "/Users/hcferguson/anaconda/envs/specviz_env/lib/python3.5/site-packages/astropy/units/core.py", line 968, in to
    return self._get_converter(other, equivalencies=equivalencies)(value)
  File "/Users/hcferguson/anaconda/envs/specviz_env/lib/python3.5/site-packages/astropy/units/core.py", line 869, in _get_converter
    self, other, self._normalize_equivalencies(equivalencies))
  File "/Users/hcferguson/anaconda/envs/specviz_env/lib/python3.5/site-packages/astropy/units/core.py", line 860, in _apply_equivalencies
    unit_str, other_str))
astropy.units.core.UnitConversionError: 'erg / (Angstrom cm2 s)' (spectral flux density wav) and 'Jy' (spectral flux density) are not convertible
Abort trap: 6
```

**Comment by nmearl** [comment]
We don't currently make any assumptions about equivalencies. But if we always expect the flux axis to be a spectral density, we can build it in.

**Comment by nmearl** [comment]
Try now with the latest commit 8b36cb156398c73dd52df9977639d0c2cffe0764.

**Comment by pllim** [comment]
It is probably safe to always include these equivalencies:

* flux - `spectral_density()`
* dispersion - `spectral()`

**Comment by nmearl** [comment]
Done in a6544909311aa4ff7ebeec318b37fc94e50cfdbb.

**Comment by pllim** [comment]
Somewhat related: What is the desired conversion behavior when user loads a spectrum with flux unit in `electrons/s` or `electrons/s/pix`? Do we just plainly disallow any unit conversion in that case? Because you cannot convert to an actual flux unit without knowing the instrument, which I think is beyond the scope of this software.

**Comment by hcferguson** [comment]
@plim in the case of HST, there is a header keyword that would allow conversion PHOTFLAM, I think. We should think about putting the capability of specifying a conversion factor in the custom loader so that people don't have to modify the file itself.

**Comment by hcferguson** [comment]
Nevermind, @plim. PHOTFLAM only exists for imaging I think. There's not a constant conversion for any realistic spectrum in electrons/s. I think disallow. At least for now. People would have to read in a calibration file to do the conversion, and that's way out of scope.

**Comment by nmearl** [comment]
The premise of this issue has been solved. The `electron/s` stuff was fixed with #166.
